File: sprint3/By languages/C#/lesson4code1.cs
using System;

public class Program {
    static void Main(string[] args) {
        GenBinary(3, "");
    }

    static void GenBinary(int n, string prefix) {
        if (n == 0) {
            Console.WriteLine(prefix);
        } else {
            GenBinary(n - 1, prefix + "0");
            GenBinary(n - 1, prefix + "1");
        }
    }
}

File: sprint3/By languages/swift/lesson4code1.swift
func genBinary(n: Int, prefix: String) {
    if n == 0 {
        print(prefix)
    } else {
        genBinary(n: n - 1, prefix: prefix + "0")
        genBinary(n: n - 1, prefix: prefix + "1")
    }
}

genBinary(n: 3, prefix: "")
